### PR TITLE
Some fixes for the networking subsystem, serverclient.c now works, some warnings fixed

### PR DESCRIPTION
**Pull request by jesings**

### CODE REVIEW

File: src/safeposix/syscalls/net_calls.rs [hunk 1]
```diff
@@ -603,7 +605,7 @@ impl Cage {
                             if let None = sockfdobj.localaddr {
                                 let bindret = sockobj.bind(&ladr);
                                 if bindret < 0 {
-                                    let sockerrno = match Errno::from_discriminant(unsafe{*libc::__errno_location()} as i32) {
+                                    match Errno::from_discriminant(unsafe{*libc::__errno_location()} as i32) {
```
Comment (rennergade) [inline src/safeposix/syscalls/net_calls.rs:608]: This seems to do some stuff that we want via the interface.

**Review by rennergade — COMMENTED** [review]
This mostly looks good, one comment.

File: src/safeposix/syscalls/net_calls.rs [hunk 2]
```diff
@@ -737,7 +739,7 @@ impl Cage {
                                 }
                             };
 
-                            if let Err(errval) = acceptedresult {
+                            if let Err(_) = acceptedresult {
                                 match Errno::from_discriminant(unsafe{*libc::__errno_location()} as i32) {
```
Comment (TristanB22) [inline src/safeposix/syscalls/net_calls.rs:743]: Same as Nick's comment above

File: src/safeposix/syscalls/net_calls.rs [hunk 3]
```diff
@@ -1318,11 +1322,6 @@ impl Cage {
                     return syscall_error(Errno::EINVAL, "epoll wait", "max events argument is not a positive number");
                 }
 
-                //now we know that all of the arguments are valid...
-                let mut reads = interface::RustHashSet::<i32>::new();
-                let mut writes = interface::RustHashSet::<i32>::new();
-                let mut errors = interface::RustHashSet::<i32>::new();
```
Comment (TristanB22) [inline src/safeposix/syscalls/net_calls.rs:1324]: I think that these are in the original system call from Repy... are they unnecessary when using Rust?

**Review by TristanB22 — APPROVED** [review]

**Review by kapkic — APPROVED** [review]
Looks good.

File: src/safeposix/syscalls/net_calls.rs (hunk 1 above)
Comment (youngfalcon) [reply src/safeposix/syscalls/net_calls.rs:608]: Agreed, since we want to restrict unsafe code to the interface.

**Review by youngfalcon — COMMENTED** [review]
This looks good in general, although moving the unsafe code portion mentioned by @rennergade and @TristanB22  might be ideal.

File: src/safeposix/syscalls/net_calls.rs (hunk 3 above)
Comment (jesings) [reply src/safeposix/syscalls/net_calls.rs:1324]: Yeah they're not actually used in poll. I think they were unused in repy as well, but probably dated back from when poll was handled separately from select in repy.

**Review by youngfalcon — APPROVED** [review]

**Comment by rennergade** [comment]
I think this is good to go now!

**Pull request merged**